crates/red_knot_workspace/tests/check.rs

```diff
@@ -272,7 +272,8 @@ impl SourceOrderVisitor<'_> for PullTypesVisitor<'_> {
 #[rustfmt::skip]
 const KNOWN_FAILURES: &[(&str, bool, bool)] = &[
     // related to circular references in class definitions
-    ("crates/ruff_linter/resources/test/fixtures/pyflakes/F821_26.py", true, false),
+    ("crates/ruff_linter/resources/test/fixtures/pyflakes/F821_26.py", true, true),
+    ("crates/ruff_linter/resources/test/fixtures/pyflakes/F821_27.py", true, true),
```

Comment:
The test failures were because we're now noticing the recursiveness of some additional class definitions (https://github.com/astral-sh/ruff/issues/14672)

```py
class Tree(list[Tree | Leaf]): ...
```

We now see into the `BitOr` of two `ClassLiterals`, and in most cases, can see via the typeshed that the result is a `UnionType`.  But in these two fixtures, there's a recursive reference inside that union, which we can't yet handle.

Reply:
Maybe eliminating the special case and just returning `types.UnionType` instead, as commented above, will help these cases?

Reply:
Oh, perhaps not; I see now that this comment was made before you added that special case.